tests/dragonfly/eval_test.py

```diff
@@ -264,3 +267,32 @@ async def test_lua_auto_async(async_client: aioredis.Redis):
 
     flushes = (await async_client.info("transaction"))["eval_squashed_flushes"]
     assert 1 <= flushes <= 3  # all 100 commands are executed in at most 3 batches
+
+
+"""
+Ensure liveness even with only a single interpreter in scenarios where EVAL and EVAL inside multi run concurrently while also contending for keys
+"""
+
+
+@dfly_args({"proactor_threads": 3, "interpreter_per_thread": 1})
```

Comment:
1. I use pytest only because I can't reinitialize the local server state with clean arguments. Maybe it makes sense to implement a C++ test because they are quicker and offer a greater load with better timing
2. I checked, it fails without the fix